### PR TITLE
Add scala serializability checks for interfaces

### PR DESCRIPTION
**Pull request by akrmn**
### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description
- [ ] If you mean to change the status of a component, please make sure you keep [the Component Status page](https://github.com/digital-asset/daml/blob/main/docs/source/support/component-statuses.rst) up to date.

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: daml-lf/validation/src/main/scala/com/digitalasset/daml/lf/validation/Serializability.scala [hunk 1]
```diff
@@ -148,6 +148,19 @@ private[validation] object Serializability {
     Env(version, interface, context, SRExceptionArg, tyCon).checkType()
   }
 
+  def checkInterface(
```
Comment (akrmn) [inline daml-lf/validation/src/main/scala/com/digitalasset/daml/lf/validation/Serializability.scala:151]: I pretty much adapted the haskell check in `compiler/daml-lf-tools/src/DA/Daml/LF/TypeChecker/Serializability.hs` to the style of the other functions in this file

File: daml-lf/validation/src/main/scala/com/digitalasset/daml/lf/validation/Serializability.scala [hunk 2]
```diff
@@ -32,9 +32,26 @@ private[validation] object Serializability {
...
+    def isInterface(typ: Type): Boolean = {
+      typ match {
+        case TTyCon(tycon) =>
+          interface.lookupDataType(tycon) match {
+            case Right(DDataType(_, _, cons)) =>
+              cons match {
+                case DataInterface => true
+                case _ => false
+              }
+            case Left(_) => false
```
Comment (akrmn) [inline daml-lf/validation/src/main/scala/com/digitalasset/daml/lf/validation/Serializability.scala:44]: the haskell `isInterface` also looks at the `currentModule`, but iiuc that only happens for serializability _inference_, whereas in scala (iiuc) we only do serializability _validation_

**Review by sofiafaro-da — APPROVED** [review]
Nice, thank you!

**Review by cocreature — COMMENTED** [review]
Looks great, let’s please add some tests to https://github.com/digital-asset/daml/blob/main/daml-lf/validation/src/test/scala/com/digitalasset/daml/lf/validation/SerializabilitySpec.scala as part of this PR.

File: daml-lf/validation/src/test/scala/com/digitalasset/daml/lf/validation/SerializabilitySpec.scala [hunk 3]
```diff
@@ -319,6 +319,33 @@ class SerializabilitySpec extends AnyWordSpec with TableDrivenPropertyChecks wit
...
 
     }
+
+    "reject unserializable interface definitions" in {
+
+      val pkg =
+        p"""
+          module NegativeTestCase {
+            interface (this: Token) = {
+              precondition False;
```
Comment (akrmn) [inline daml-lf/validation/src/test/scala/com/digitalasset/daml/lf/validation/SerializabilitySpec.scala:329]: took me a bit to figure it out but the `precondition` component turned out to be mandatory

**Pull request merged**